Fix some remaining bashism issues is safe_asterisk.
Fixes, or mostly fixes, #530047.

Upstream issue: https://issues.asterisk.org/view.php?id=10812

--- a/contrib/scripts/safe_asterisk
+++ b/contrib/scripts/safe_asterisk
@@ -62,7 +62,7 @@ then
 	PRIORITY=0
 	message "safe_asterisk was started by `id -n` (uid `id -u`)."
 else
-	if `echo $OSTYPE | grep linux 2>&1 > /dev/null `
+	if `uname -s | grep Linux 2>&1 > /dev/null `
 	then
 		# maximum number of open files is set to the system maximum divided by two if
 		# MAXFILES is not set.
@@ -75,7 +75,7 @@ else
 			fi
 		fi
 		SYSCTL_MAXFILES="fs.file-max"
-	elif `echo $OSTYPE | grep darwin 2>&1 > /dev/null `
+	elif `uname -s | grep Darwin 2>&1 > /dev/null `
 	then
 		SYSCTL_MAXFILES="kern.maxfiles"
 	fi
@@ -143,7 +143,7 @@ trap '' PIPE
 if [ -d /etc/asterisk/startup.d ]; then
 	for script in /etc/asterisk/startup.d/*.sh; do
 		if [ -x ${script} ]; then
-			source ${script}
+			. ${script}
 		fi
 	done
 fi
@@ -167,7 +167,7 @@ run_asterisk()
 			message "Asterisk shutdown normally."
 			exit 0
 		elif [ $EXITSTATUS -gt 128 ]; then
-			let EXITSIGNAL=EXITSTATUS-128
+			EXITSIGNAL=EXITSTATUS-128
 			echo "Asterisk exited on signal $EXITSIGNAL."
 			if [ "$NOTIFY" != "" ]; then
 				echo "Asterisk on $MACHINE exited on signal $EXITSIGNAL.  Might want to take a peek." | \
